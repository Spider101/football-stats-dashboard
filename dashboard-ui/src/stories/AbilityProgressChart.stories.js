import AbilityProgressChart from '../components/AbilityProgressChart';
import { getPlayerAbilityData } from './utils/storyDataGenerators';

export default {
    component: AbilityProgressChart,
    title: 'Components/PlayerProgressionView/AbilityProgressChart',
    excludeStories: /.*Data$/,
    argTypes: {
        abilityData: {
            name: 'Ability Progression Data',
            control: { type: 'object' }
        }
    },
    parameters: {
        docs: {
            description: {
                component: 'UI Component for displaying a bar chart documenting a player\'s Ability progress over the'
                + ' last _N_ months.'
            }
        }
    }
};

const Template = args => <AbilityProgressChart { ...args } />;

export const MultipleDataPoints = Template.bind({});
MultipleDataPoints.args = {
    abilityData: getPlayerAbilityData()
};

export const SingleDataPoint = Template.bind({});
SingleDataPoint.args = {
    abilityData: getPlayerAbilityData(1)
};