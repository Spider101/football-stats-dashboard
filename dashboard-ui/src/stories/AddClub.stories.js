import AddClub from '../widgets/AddClub';

export default {
    component: AddClub,
    title: 'Widgets/ClubPageView/AddClub',
    argTypes: { addClubAction: { control: { disable: true } } },
    decorators: [
        Story => (
            <div style={{ display: 'flex', justifyContent: 'center' }}>
                <Story />
            </div>
        )
    ]
};

const Template = args => <AddClub {...args} />;

export const Success = Template.bind({});
Success.args = {
    addClubAction: _ => null
};

export const FormFailure = Template.bind({});
FormFailure.args = {
    addClubAction: _ => 'Failed to add new club!'
};