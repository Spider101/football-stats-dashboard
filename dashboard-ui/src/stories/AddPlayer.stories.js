import AddPlayer from '../widgets/AddPlayer';

export default {
    component: AddPlayer,
    title: 'Widgets/SquadHubView/AddPlayer',
    argTypes: { addPlayerAction: { control: { disable: true } } },
    decorators: [
        Story => (
            <div style={{ display: 'flex', justifyContent: 'center' }}>
                <Story />
            </div>
        )
    ]
};

const Template = args => <AddPlayer {...args} />;

export const Success = Template.bind({});
Success.args = {
    addPlayerAction: _ => null
};

export const FormFailure = Template.bind({});
FormFailure.args = {
    addPlayerAction: _ => 'Failed to add new player!'
};