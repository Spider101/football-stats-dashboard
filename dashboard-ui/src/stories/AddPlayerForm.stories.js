import { action } from '@storybook/addon-actions';
import { LookupDataContextProvider } from '../context/LookupDataProvider';
import { getLookupDataHandlers } from '../mocks/handlers';

import AddPlayerForm from '../widgets/AddPlayerForm';

export default {
    component: AddPlayerForm,
    title: 'Widgets/SquadHubView/AddPlayerForm',
    decorators: [
        Story => (
            <LookupDataContextProvider>
                <Story />
            </LookupDataContextProvider>
        )
    ]
};

const getFormMetadataAtStep = () => ({
    formData: { name: '', age: '', country: '' },
    formValidations: {},
    handleChangeFn: action('field was changed')
});

const Template = args => <AddPlayerForm {...args} />;
export const Default = Template.bind({});
Default.args = {
    getFormMetadataAtStep,
    stepIdx: 0
};

Default.parameters = {
    msw: {
        handlers: getLookupDataHandlers()
    }
};