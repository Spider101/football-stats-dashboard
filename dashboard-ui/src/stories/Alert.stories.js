import React from 'react';

import Alert from '../components/Alert';

export default {
    component: Alert,
    title: 'Components/Globals/Alert'
};

const Template = args => <Alert { ...args } />;

export const Success = Template.bind({});
Success.args = {
    severity: 'success',
    text: 'This is a success message!'
};

export const Error = Template.bind({});
Error.args = {
    severity: 'error',
    text: 'This is an error message!'
};