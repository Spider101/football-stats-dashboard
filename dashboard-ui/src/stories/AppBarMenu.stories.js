import React from 'react';
import { action } from '@storybook/addon-actions';

import AppBarMenu from '../components/AppBarMenu';
import { AuthContextProvider } from '../context/authProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

export default {
    component: AppBarMenu,
    title: 'Components/Globals/AppBarMenu',
    parameters: {
        docs: {
            description: {
                component: 'UI Component at the top of the application for displaying information pertinent to the'
                + ' current screen. Currently, it hosts functionality like displaying the team\'s name, a search bar'
                + 'and a user profile badge UI element. It also contains a button for expanding and closing'
                + 'the side bar menu.'
            }
        }
    },
    decorators: [
        Story => (
            <QueryClientProvider client={new QueryClient()}>
                <AuthContextProvider>
                    <Story />
                </AuthContextProvider>
            </QueryClientProvider>
        )
    ]
};

const Template = args => <AppBarMenu {...args} />;

export const Default = Template.bind({});
Default.args = {
    menu: {
        title: 'Manchester United',
        teamColor: 'red'
    },
    onClickHandler: action('open-menu'),
    isOpen: false
};

export const MenuOpened = Template.bind({});
MenuOpened.args = {
    ...Default.args,
    isOpen: true
};
