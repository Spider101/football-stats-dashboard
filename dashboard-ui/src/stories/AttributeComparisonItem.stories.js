import React from 'react';
import AttributeComparisonItem from '../components/AttributeComparisonItem';
import faker from 'faker';

export const getAttrComparisonItemData = () => ({
    attrComparisonItem: {
        attrValues: [{
            name: faker.name.lastName(1),
            data: [ Math.round(Math.random() * 20) ]
        }, {
            name: faker.name.lastName(1),
            data: [ -1 * Math.round(Math.random() * 20) ]
        }],
        label: faker.hacker.noun()
    },
    isHighlighted: false
});

export default {
    component: AttributeComparisonItem,
    title: 'Components | PlayerComparisonView/AttributeComparisonTable/AttributeComparisonItem',
    excludeStories: /.*Data$/,
};

export const Default = () => <AttributeComparisonItem { ...getAttrComparisonItemData() } />;

export const Highlighted = () => <AttributeComparisonItem { ...getAttrComparisonItemData() } isHighlighted={ true }/>;
