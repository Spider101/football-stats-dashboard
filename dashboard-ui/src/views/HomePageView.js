import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { makeStyles } from '@material-ui/core/styles';

import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles(theme => ({
    fab: {
        position: 'fixed',
        bottom: theme.spacing(4),
        right: theme.spacing(4)
    },
    clubs: {
        margin: theme.spacing(2)
    }
}));

export default function HomePageView({ clubs, addClubWidget }) {
    const classes = useStyles();

    const noClubsView = (
        <Typography variant='h5' style={{ textAlign: 'center', padding: '5%' }}>
            No clubs have been created yet! Please create a club to proceed.
        </Typography>
    );

    const allCLubsView = (
        <List className={classes.clubs}>
            {clubs.map(club => {
                return (
                    <ListItem key={club.id} button divider component={Link} to={`/club/${club.id}`}>
                        <ListItemText primary={club.name} />
                    </ListItem>
                );
            })}
        </List>
    );

    return (
        <Container>
            <Typography variant='h2'>Clubs</Typography>
            <Divider />
            {clubs.length === 0 ? noClubsView : allCLubsView}
            <div className={classes.fab}>
                {addClubWidget}
            </div>
        </Container>
    );
}

HomePageView.propTypes = {
    clubs: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            transferBudget: PropTypes.string,
            wageBudget: PropTypes.string,
            income: PropTypes.string,
            expenditure: PropTypes.string
        })
    ),
    addClubWidget: PropTypes.node
};