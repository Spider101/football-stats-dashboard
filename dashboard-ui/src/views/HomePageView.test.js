import { render, screen } from '@testing-library/react';

import { Default, NoClubs } from '../stories/HomePageView.stories';
import { MemoryRouter } from 'react-router';

const noClubViewText = 'No clubs have been created yet! Please create a club to proceed.';
it('should render the club data passed in', () => {
    render(
        <MemoryRouter>
            <Default {...Default.args} />
        </MemoryRouter>
    );

    expect(screen.queryByText(noClubViewText)).not.toBeInTheDocument();
    Default.args.clubs.map(club => {
        const clubItem = screen.getByRole('button', { name: club.name });
        expect(clubItem).toHaveAttribute('href', `/club/${club.id}`);
    });
});

it('should render helpful text when no club data is passed in', () => {
    render(
        <MemoryRouter initialEntries={['/']}>
            <NoClubs {...NoClubs.args} />
        </MemoryRouter>
    );
    expect(screen.queryByText(noClubViewText)).toBeInTheDocument();
});

it('should always render add club widget', () => {
    const { rerender, container } = render(
        <MemoryRouter initialEntries={['/']}>
            <NoClubs {...NoClubs.args} />
        </MemoryRouter>
    );
    expect(screen.queryByLabelText('add')).toBeInTheDocument();

    // taking snapshot here mainly to track any breaking changes due to add club widget
    expect(container).toMatchSnapshot();

    // not taking snapshot here as club props are dynamically generated for test
    rerender(
        <MemoryRouter initialEntries={['/']}>
            <Default {...Default.args} />
        </MemoryRouter>
    );
    expect(screen.queryByLabelText('add')).toBeInTheDocument();
});