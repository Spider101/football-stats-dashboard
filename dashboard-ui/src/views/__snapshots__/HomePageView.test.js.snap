// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should always render add club widget 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      Clubs
    </h2>
    <hr
      class="MuiDivider-root"
    />
    <h5
      class="MuiTypography-root MuiTypography-h5"
      style="text-align: center; padding: 5%;"
    >
      No clubs have been created yet! Please create a club to proceed.
    </h5>
    <div
      class="makeStyles-fab-9"
    >
      <button
        aria-label="add"
        class="MuiButtonBase-root MuiFab-root MuiFab-secondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
